lib/output/cassandra.go

```diff
@@ -0,0 +1,33 @@
+package output
+
+import (
+	"github.com/Jeffail/benthos/v3/lib/log"
+	"github.com/Jeffail/benthos/v3/lib/metrics"
+	"github.com/Jeffail/benthos/v3/lib/output/writer"
+	"github.com/Jeffail/benthos/v3/lib/types"
+)
+
+//------------------------------------------------------------------------------
+
+func init() {
+	Constructors[TypeCassandra] = TypeSpec{
+		constructor: NewCassandra,
+		Description: `
+Sends messages to a Cassandra database.`,
+	}
+}
+
+//------------------------------------------------------------------------------
+
+// NewCassandra creates a new Cassandra output type.
+func NewCassandra(conf Config, mgr types.Manager, log log.Modular, stats metrics.Type) (Type, error) {
+	c, err := writer.NewCassandra(conf.Cassandra, log, stats)
+	if err != nil {
+		return nil, err
+	}
+	return NewWriter(
```

Comment:
If we use `NewAsyncWriter` here instead then we can replace the `async` parameter with `max_in_flight` similar to other outputs like nats: https://github.com/Jeffail/benthos/blob/master/lib/output/nats_stream.go#L43, although you don't need to bother with the check for `1` just always create a `NewAsyncWriter` with whatever the field is set to.